src/worker/builder.rs

```diff
@@ -152,15 +152,19 @@ impl<E: 'static> WorkerBuilder<E> {
     }
 
     /// Connect to a Faktory server with a non-standard stream.
-    pub async fn connect_with<S: AsyncRead + AsyncWrite + Send + Unpin>(
+    pub async fn connect_with<S>(
         mut self,
         stream: S,
         pwd: Option<String>,
-    ) -> Result<Worker<BufStream<S>, E>, Error> {
+    ) -> Result<Worker<E>, Error>
+    where
+        S: AsyncRead + AsyncWrite + Send + Sync + Unpin + 'static,
+        BufStream<S>: Reconnect,
+    {
         self.opts.password = pwd;
         self.opts.is_worker = true;
         let buffered = BufStream::new(stream);
```

Comment:
How do you feel about _not_ wrapping the provided stream in `BufStream` here and instead encouraging the caller to do so? My thinking is that they may _already_ have some stream type that does buffering, and we then force them here to have _two_ layers of buffering.

Reply:
My thinking was that requiring `AsyncBufRead` is not something standard/predictable, this is why we only ask them for AsyncRead/AsyncWrite. But I also agree that if they already have got a buffered one - maybe it makes sense to offer them an optional way. Looking into it

Reply:
Added `connect_with_buffered` for both `Client` and `WorkerBuilder` with the according docs. Also updated one of the tests to use this method just as a sanity check. The ordinary `connect_with` will be  performing the buffering internally and passing the stuff through to `connect_with_buffered`, wdyt?

Reply:
I think I actually prefer just having a single `connect_with` and noting in its docs that you should probably wrap the passed-in stream in `BufStream::new` if it doesn't already implement `AsyncBufRead`. After all, most folks will only use this method if they specifically have a custom stream already!

Reply:
updated